great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -234,9 +244,17 @@ def __init__(self, table_name=None, engine=None, connection_string=None,
             # a user-defined schema
             raise ValueError("Cannot specify both schema and custom_sql.")
 
+        if custom_sql is not None and self.engine.dialect.name.lower() == "bigquery":
+            if generated_table_name is not None and self.engine.dialect.dataset_id is None:
+                raise ValueError("No BigQuery dataset specified. Use biquery_temp_table batch_kwarg or a specify a default dataset in engine url")
+
         if custom_sql:
             self.create_temporary_table(table_name, custom_sql)
 
+            if generated_table_name is not None:
+                logger.warn("Created permanent table {table_name}".format(
```

Comment:
stylistic consistency...

```suggestion
                logger.warning("Created permanent table {table_name}".format(
```